SimMuon/MCTruth/src/CSCHitAssociator.cc

```diff
@@ -0,0 +1,108 @@
+#include "SimMuon/MCTruth/interface/CSCHitAssociator.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "SimDataFormats/TrackerDigiSimLink/interface/StripDigiSimLink.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "Geometry/Records/interface/MuonGeometryRecord.h"
+
+CSCHitAssociator::CSCHitAssociator(const edm::Event& event, const edm::EventSetup& setup, const edm::ParameterSet& conf): 
+  theDigiSimLinks(0),
+  linksTag(conf.getParameter<edm::InputTag>("CSClinksTag"))
+{
+  initEvent(event,setup);
+}
+
+CSCHitAssociator::CSCHitAssociator( const edm::ParameterSet& conf, edm::ConsumesCollector && iC ): 
+  theDigiSimLinks(0),
+  linksTag(conf.getParameter<edm::InputTag>("CSClinksTag"))
+{
+  iC.consumes<DigiSimLinks>(linksTag);
+}
+
+void CSCHitAssociator::initEvent(const edm::Event& event, const edm::EventSetup& setup) {
+
+  edm::Handle<DigiSimLinks> digiSimLinks;
+  LogTrace("CSCHitAssociator") <<"getting CSC Strip DigiSimLink collection - "<<linksTag;
+  event.getByLabel(linksTag, digiSimLinks);
+  theDigiSimLinks = digiSimLinks.product();
+
+  // get CSC Geometry to use CSCLayer methods
+  edm::ESHandle<CSCGeometry> mugeom;
+  setup.get<MuonGeometryRecord>().get( mugeom );
+  cscgeom = &*mugeom;
```

Comment:
>the new code lost a functionality which would be still useful for run1 scenario, related to the so-called "bad" CSC chambers, i.e. the ones existing in geometry but not providing hits in DIGI-RECO.
see the current code:
https://github.com/cms-sw/cmssw/blob/CMSSW_7_4_X/SimMuon/MCTruth/interface/MuonTruth.h#L51
https://github.com/cms-sw/cmssw/blob/CMSSW_7_4_X/SimMuon/MCTruth/src/MuonTruth.cc#L60 -L63